### PR TITLE
How to build a Meme Generator with Vanilla JavaScript, Skeleton and the HTML 5 Canvas

### LINKED ISSUE #5765: How to build a Meme Generator with Vanilla JavaScript, Skeleton and the HTML 5 Canvas.
### Writing sample(s): 
https://stackoverflow.blog/2021/10/20/why-hooks-are-the-best-thing-to-happen-to-react/?cb=1&_ga=2.218645726.618497180.1634907321-1032916238.1629601197

https://www.section.io/engineering-education/how-to-build-a-currency-converter-with-react-and-its-material-ui/

https://www.section.io/engineering-education/building-a-calculator-a-javascript-project-for-beginners/

#### Proposed title of the article
How to build a Meme Generator with Vanilla JavaScript, Skeleton and the HTML 5 Canvas. 

#### Proposed article introduction
Memes are a stimulating and fun way to pass messages across to people. In this article, we will take a step-by-step guide on creating a Meme Generator with pure Vanilla JavaScript and style its components with [Skeleton](http://getskeleton.com/). With the use of the HTML 5 Canvas, users will be able to use images and texts of their choice locally and or from an online source to create comical memes for a mass online audience. It is a must-do JavaScript project for beginners.

#### Key takeaways 
1. Readers will learn how to import images to this application both locally and from an online source and use them to create Memes.
2. Readers will learn how to use Skeleton(a dead simple and responsive boilerplate to kickstart any responsive project) to style any application.
3. Readers will learn how to use the HTML 5 Canvas to update the memes created in the application. 

#### Article quality
This is an in-depth article targeted at beginner JavaScript developers. This tutorial will cover the entire process of importing Skeleton to your application and using it to design the UI. It also provides handy tips on how to go about using the HTML 5 Canvas in your JavaScript project. 

#### References
N/A

### PR DESCRIPTION
**Pull request by Nomzy-kush**
## Our **Peer Reviewer and Content Moderator teams do NOT provide any revisions services.** All revisions and edits should be completed on your own forked repo (as to not take up room in the queue) in order for our team to review them all in a timely manner. 

**ANY ARTICLE SUBMITTED WITH GLARING ERRORS WILL BE IMMEDIATELY CLOSED.**

>As a rule of thumb - please be sure to only submit articles (pull requests) that are fully polished and ready to be published. Be sure to go through our resources documents for extra and 3rd party (vetted) resources to help improve overall technical writing.

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This fixes #5765

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by Jeff-mwangi** [comment]
@hectorkambow and @ahmadmardeni1 I would like to review this article

**Comment by ahmadmardeni1** [comment]
Go ahead. @Jeff-mwangi

**Comment by ahmadmardeni1** [comment]
Upon running your article through our 3rd party plagiarism checker it seemed to raise a few flags and the % was higher than we typically accept.
Please see attached PDF - and revisit the article to ensure we are contributing wholly unique and original content.
Be sure to see our resources page to see more info on plagiarism and what is considered as such.

**Comment by ahmadmardeni1** [comment]
Upon running your article through our 3rd party plagiarism checker it seemed to raise a few flags and the % was higher than we typically accept. @Nomzy-kush 
Please see attached PDF - and revisit the article to ensure we are contributing wholly unique and original content.
Be sure to see our resources page to see more info on plagiarism and what is considered as such.


[How to build a Meme Generator with Vanilla JavaScript, Skeleton and the HTML 5 Canvas#6172](https://github.com/section-engineering-education/engineering-education/files/7920025/New.Scan.7_57.AM.Copyleaks.Report.1.pdf)

**Comment by Nomzy-kush** [comment]
>
> Hey Jeff,
>
Upon running the draft through several Plagiarism checkers, I’ve only seen
little to no plagiarism. I was wondering if you can send me the link to the
Plagiarism report you got.

Thanks,
Onome

**Comment by Nomzy-kush** [comment]
Hello @ahmadmardeni1 you can go ahead to recheck this draft again for plagiarism.

**Comment by ahmadmardeni1** [comment]
Plagiarism check done ✅ @Nomzy-kush

File: content/articles/how-to-build-a meme-generator-with-vanilla-javascript-skeleton-and-the-html5-canvas/index.md [hunk 1]
```diff
@@ -0,0 +1,313 @@
+# How to build a Meme Generator with Vanilla JavaScript, Skeleton and the HTML 5 Canvas
+Memes are a stimulating and fun way to pass messages across to people. This tutorial will take you through a step-by-step guide on how you can build a simple meme generator with JavaScript(no frameworks involved) and style its components with CSS and [Skeleton](http://getskeleton.com/). With the [HTML 5 Canvas](https://developer.mozilla.org/en-US/docs/Web/API/Canvas_API), users will be able to upload images and texts of their choice and or from an online source to create comical memes for a mass online audience.
```
Comment (Jeff-mwangi) [inline content/articles/how-to-build-a meme-generator-with-vanilla-javascript-skeleton-and-the-html5-canvas/index.md:2]:
```suggestion
Memes are a stimulating and fun way to pass messages to people. This tutorial will take you through a step-by-step guide on how you can build a simple meme generator with JavaScript(no frameworks involved) and style its components with CSS and [Skeleton](http://getskeleton.com/). With the [HTML 5 Canvas](https://developer.mozilla.org/en-US/docs/Web/API/Canvas_API), users will be able to upload images and texts of their choice and or from an online source to create comical memes for a mass online audience. 
```

File: content/articles/how-to-build-a meme-generator-with-vanilla-javascript-skeleton-and-the-html5-canvas/index.md [hunk 2]
```diff
@@ -0,0 +1,313 @@
+# How to build a Meme Generator with Vanilla JavaScript, Skeleton and the HTML 5 Canvas
+Memes are a stimulating and fun way to pass messages across to people. This tutorial will take you through a step-by-step guide on how you can build a simple meme generator with JavaScript(no frameworks involved) and style its components with CSS and [Skeleton](http://getskeleton.com/). With the [HTML 5 Canvas](https://developer.mozilla.org/en-US/docs/Web/API/Canvas_API), users will be able to upload images and texts of their choice and or from an online source to create comical memes for a mass online audience. 
+
+## Prerequisites for this tutorial
```
Comment (Jeff-mwangi) [inline content/articles/how-to-build-a meme-generator-with-vanilla-javascript-skeleton-and-the-html5-canvas/index.md:4]: Add table of contents

**Comment by Jeff-mwangi** [comment]
Hello @Nomzy-kush Consider doing all the changes first and then commit them once.

**Comment by Nomzy-kush** [comment]
Hey @Jeff-mwangi I have made all the necessary changes you requested.

**Comment by hectorkambow** [comment]
@Jeff-mwangi 

If any more edits are needed, we will be closing this PR 👍

**Comment by hectorkambow** [comment]
@jeff-mwangi any updates on this?

**Comment by Jeff-mwangi** [comment]
> @Jeff-mwangi any updates on this?

Working on tech review

File: content/articles/how-to-build-a meme-generator-with-vanilla-javascript-skeleton-and-the-html5-canvas/index.md [hunk 3]
```diff
@@ -0,0 +1,346 @@
+### How to build a Meme Generator with Vanilla JavaScript, Skeleton and the HTML 5 Canvas
```
Comment (Jeff-mwangi) [inline content/articles/how-to-build-a meme-generator-with-vanilla-javascript-skeleton-and-the-html5-canvas/index.md:1]: Add front matter
```suggestion
---
layout: engineering-education
status: publish
published: true
url: /how-to-build-a meme-generator-with-vanilla-javascript-skeleton-and-the-html5-canvas/
title: Building a meme generator with Vanilla JavaScript, Skeleton and the HTML 5 Canvas
description: This tutorial will show you how to build a meme generator with Vanilla JavaScript, Skeleton and the HTML 5 Canvas
author: doro-onome
date: 2022-02-09T00:00:00-22:15
topics: [Languages]
excerpt_separator: <!--more-->
images:

  - url: /engineering-education/how-to-build-a meme-generator-with-vanilla-javascript-skeleton-and-the-html5-canvas/hero.jpg
    alt: Building a meme generator with Vanilla JavaScript, Skeleton and the HTML 5 Canvas
---
```

**Review by Jeff-mwangi — CHANGES_REQUESTED** [review]
@Nomzy-kush Add front matter and PRC line.

**Comment by louisefindlay23** [comment]
Please link your topic suggestion so we can accurately pay out if this article is published

**Comment by hectorkambow** [comment]
> Please link your topic suggestion so we can accurately pay out if this article is published

🤔  

Not sure how the article topic is connected to the payout (unless im misunderstanding) - important step nonetheless

**Comment by hectorkambow** [comment]
> > @Jeff-mwangi any updates on this?
> 
> Working on tech review

Thank you - lets try to keep our reviews (wait time to a minimum if possible - we understand things come up but) to be fair to our authors and other in the queue.

**Comment by hectorkambow** [comment]
@Jeff-mwangi  

If any other edits are needed - we will be closing the PR - this has underwent an extensive review already 🚀

**Comment by Nomzy-kush** [comment]
> Please link your topic suggestion so we can accurately pay out if this article is published

This fixes #5765

**Review by Jeff-mwangi — APPROVED** [review]
Thank you @Nomzy-kush . The changes looks good now. Let's move it for the final review.

**Comment by Jeff-mwangi** [comment]
@ahmadmardeni1 This article is now ready for final review.

cc: @hectorkambow

**Comment by Nomzy-kush** [comment]
Hey @hectorkambow and @louisefindlay23, isn't this PR ready to be merged?

**Comment by hectorkambow** [comment]
jumping on this one

**Comment by hectorkambow** [comment]
@section-engineering-education/content-moderator-team 
We should have pushed this article along as well. 👀

**Pull request merged**